spec/jobs/deposit_job_spec.rb

```diff
@@ -181,9 +215,94 @@
           .with(a_hash_including(version_description: 'Changed files')) do |params|
           external_identifier = params[:metadata].structural.contains.first.structural.contains.first.externalIdentifier
+          size = params[:metadata].structural.contains.first.structural.contains.first.size
+          has_mime_type = params[:metadata].structural.contains.first.structural.contains.first.hasMimeType
           expect(external_identifier).to eq('9999999')
+          expect(size).to eq(17_675)
+          expect(has_mime_type).to eq('image/svg+xml')
         end
+        expect(SdrClient::Deposit::UploadFiles).to have_received(:upload)
+      end
+    end
 
+    context 'when a file is added to an existing deposit' do
+      let(:attached_file2) { build(:attached_file) }
+      let(:second_work_version) do
+        build(:work_version, work:, attached_files: [attached_file, attached_file2], version: 2,
+                             version_description: 'Added file')
+      end
+
+      before do
+        work.work_versions = [first_work_version, second_work_version]
+        allow(SdrClient::Find).to receive(:run).and_return(cocina.to_json)
+        allow(SdrClient::Deposit::UploadFiles).to receive(:upload)
+          .and_return([SdrClient::Deposit::Files::DirectUploadResponse.new(filename: 'sul2.svg',
+                                                                           signed_id: '9999999')])
+      end
+
+      it 'uploads files and calls UpdateResource.run' do
+        described_class.perform_now(second_work_version)
+        expect(SdrClient::Deposit::UpdateResource).to have_received(:run)
+          .with(a_hash_including(version_description: 'Added file')) do |params|
+          # file from version1
+          external_identifier = params[:metadata].structural.contains.first.structural.contains.first.externalIdentifier
+          size = params[:metadata].structural.contains.first.structural.contains.first.size
+          has_mime_type = params[:metadata].structural.contains.first.structural.contains.first.hasMimeType
+          expect(external_identifier).to eq('https://cocina.sul.stanford.edu/file/bk123gh4567-123456/sul.svg')
+          expect(size).to eq(123)
+          expect(has_mime_type).to eq('text/html')
+          # new file
+          external_identifier2 =
+            params[:metadata].structural.contains.second.structural.contains.first.externalIdentifier
+          size2 = params[:metadata].structural.contains.second.structural.contains.first.size
+          has_mime_type2 = params[:metadata].structural.contains.second.structural.contains.first.hasMimeType
+          expect(external_identifier2).to eq('9999999')
+          expect(size2).to eq(17_675)
+          expect(has_mime_type2).to eq('image/svg+xml')
+        end
+        expect(SdrClient::Deposit::UploadFiles).to have_received(:upload)
+      end
+    end
+
+    context 'when a file with the same name is replaced on a deposit' do
```

Comment:
@justinlittman I added this test to try to check the scenario of replacing a file with one with the same name. Would you take a look?